packages/babel-core/test/pattern-to-regex.js

```diff
@@ -0,0 +1,45 @@
+import path from "path";
+import patternToRegex from "../lib/config/pattern-to-regex";
+
+describe("patternToRegex", function () {
+  const dirname = process.cwd();
+
+  [
+    { pattern: ".", test: dirname, matchPrefix: true, matches: true },
+    {
+      pattern: ".",
+      test: dirname + path.sep,
+      matchPrefix: true,
+      matches: true,
+    },
+    {
+      pattern: ".",
+      test: path.resolve(dirname, "node_modules", "my-package"),
+      matchPrefix: true,
+      matches: true,
+    },
+    { pattern: ".", test: dirname, matchPrefix: false, matches: true },
+    {
+      pattern: ".",
+      test: dirname + path.sep,
+      matchPrefix: false,
+      matches: true,
+    },
+    {
+      pattern: ".",
+      test: path.join(dirname, "node_modules", "my-package"),
+      matchPrefix: false,
+      matches: false,
+    },
+  ].forEach(function ({ pattern, test, matchPrefix, matches }) {
+    it(`${test} ${matches ? "matches" : "does not match"} pattern ${pattern} when matchPrefix is ${matchPrefix}`, function () {
```

Comment:
Nit: you can use `jest.each` here instead of manual `.forEach`/`it`

Reply:
Updated this.